doc/adrs/003_add_support_for_unsynchronised_diagrams.adoc

```diff
@@ -0,0 +1,80 @@
+= ADR-003 - Add support for unsynchronized diagrams
+
+== Context
+
+Sirius RCP supports unsynschronized diagrams in which a diagram can have some of its mappings defined as unsynchronized.
+An unsynchronized mapping will not automatically trigger the appearance of a figure in the diagram if a semantic element compatible with the mapping is found.
+The figure will only appear if some tool has explicitely requested the creation of a view for this mapping and semantic element thanks to the `Create View` model operation.
+In a similar fashion, it is possible to delete a figure from the diagram without having to delete its semantic element by using the `Delete View` model operation.
+
+Sirius RCP also supports unsynchronizable mappings which, contrary to regular unsynchronized mappings, give the end users some control over the synchronization policy.
+
+We want to support unsynchronized descriptions in Sirius Components.
+In order to refresh unsynchronised descriptions in a diagram, we need to know which diagram elements have been previously created for unsynchronized descriptions in order to keep them on screen.
+This ADR is thus built on top of the previous ADR which introduces the ability to create a diagram using an existing diagram to perform an incremental refresh.
+
+
+== Decision
+
+Node and edge descriptions will now contain a synchronization policy.
+This policy will support synchronized descriptions, the default behavior, and unsynchronized descriptions.
+
+The diagram renderer will need to use some information from a previous diagram in order to keep the existing figure created from unsynchronized descriptions.
+This information will be useful for the `DiagramComponent`, which initializes the creation of the top level nodes and the edges, the `NodeComponent` which creates the various nodes and finally the `EdgeComponent` which creates the various edges.
+Since this information is only relevant for the components themselves and not for the providers such as the semantic element providers or the target object id provider, it will only appear in the props of the various components and not in the variable manager.
+
+When we will refresh a diagram using a previous diagram, if we encounter an unsynchronized description, we should only refresh the old elements created from the description and not try to create new ones.
+For that, we will need to give the `NodeComponent` and `EdgeComponent` the list of nodes and edges respectively that have been created in the previous diagram for their current node or edge description.
+In case of synchronized description, that information may also be relevant since it will give us the ability to keep layout data or a custom user-defined style from an old node or edge while creating the new one.
```

Comment:
See my other comment in `NodeComponent`: it seem the new design indeed allows for that, but the render code does not take advantage of it to implement what is mentioned here.

Reply:
Yes, that's on purpose, this PR, which is already big enough, is just for the introduction of some new raw APIs. There is still a lot to be done.